crates/marketplace/src/testing/order_test.rs

```diff
@@ -234,23 +224,20 @@ async fn test_builder_order() {
             // get transactions submitted in previous rounds, [] for genesis
             // and simulate the block built from those
             let transactions = prev_proposed_transactions.take().unwrap_or_default();
-            let (quorum_proposal, quorum_proposal_msg, da_proposal_msg, builder_state_id) =
-                calc_proposal_msg(NUM_STORAGE_NODES, round, prev_quorum_proposal, transactions)
-                    .await;
+            let (quorum_proposal, events, builder_state_id) =
+                calc_proposal_events(round, prev_quorum_proposal, transactions).await;
 
             prev_quorum_proposal = Some(quorum_proposal.clone());
 
-            // send quorum and DA proposals for this round
-            senders
-                .da_proposal
-                .broadcast(MessageType::DaProposalMessage(da_proposal_msg))
-                .await
-                .unwrap();
-            senders
-                .quorum_proposal
-                .broadcast(MessageType::QuorumProposalMessage(quorum_proposal_msg))
-                .await
-                .unwrap();
+            for evt in events {
+                event_stream_sender
+                    .broadcast(Event {
+                        view_number: ViewNumber::new(round as u64),
+                        event: evt,
+                    })
+                    .await
+                    .unwrap();
+            }
```

Comment:
Nit: This seems to be the same in several places. Can we move the `broadcast` calls to the `calc_proposal_events` function (also rename the function)?

Reply:
Sure! I've gone a step further and moved most of the proposal wrangling code to a separate type that keeps track of chain state.

Reply:
Good change!